### PR TITLE
Added code to add/remove unregistered user to/from db

### PR DESCRIPTION
**Pull request by AndreB065**
I modified the firebase user dart file to allow unregistered users to be added or removed to or from the database using a separate collection.

### CODE REVIEW

File: thc
Comment (nate-thegrate) [inline thc:1]: There was this really strange submodule; no idea how it got here. Luckily I eventually figured out how to remove it.

File: lib/firebase/user.dart [hunk 1]
```diff
@@ -147,7 +148,13 @@ sealed class ThcUser {
       };
 
   /// Saves the current user data to Firebase.
-  Future<void> upload() async => db.doc('users/$id').set(json);
+  Future<void> upload({bool isRegistered = true}) async {
+    final collection = isRegistered ? 'users' : 'unregistered_users';
+    await db.doc('$collection/$id').set(json);
+  }
```
Comment (nate-thegrate) [inline lib/firebase/user.dart:154]: Looks fantastic!

**Review by nate-thegrate — APPROVED** [review]
Thanks so much for this addition! (And for being patient with me on Slack 😄)

**Comment by AndreB065** [comment]
Thanks for the assist Nate.  Couldn't have gotten this far without you.

**Pull request merged**